test/1_stdlib/ErrorHandling.swift

```diff
@@ -340,4 +340,42 @@ ErrorHandlingTests.test("ErrorHandling/Collection map") {
   }
 }
 
+/* when b == 2 we are losing elements. insertion sort throws error
+in between while sorting so loop is not able to complete and swap the elements and array mutates
+to [2,3,3]
+ErrorHandlingTests.test("ErrorHandling/sort") {
+  var collection = [3, 2, 1]
+  do {
+    try collection.sort { (a, b) throws -> Bool in
+      print("collection before error \(collection)")
+      print("a \(a) \(b)")
+
+      if b == 2 {
```

Comment:
If it's not worth designing for, I'm not sure why it's worth implementing. If it's there, someone will start depending on it.